python/hopsworks_common/core/job_api.py

```diff
@@ -0,0 +1,273 @@
+        """
+        _client = client.get_instance()
+
+        config = util.validate_job_conf(config, _client._project_name)
+
+        path_params = ["project", _client._project_id, "jobs", name]
+
+        headers = {"content-type": "application/json"}
+        created_job = job.Job.from_response_json(
+            _client._send_request(
+                "PUT", path_params, headers=headers, data=json.dumps(config)
+            )
+        )
+        print(f"Job created successfully, explore it at {created_job.get_url()}")
+        return created_job
+
+    def get_job(self, name: str):
+        """Get a job.
+
+        # Arguments
+            name: Name of the job.
+        # Returns
+            `Job`: The Job object
+        # Raises
+            `RestAPIError`: If unable to get the job
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "jobs",
+            name,
+        ]
+        query_params = {"expand": ["creator"]}
+        return job.Job.from_response_json(
+            _client._send_request("GET", path_params, query_params=query_params)
+        )
+
+    def get_jobs(self):
+        """Get all jobs.
+
+        # Returns
+            `List[Job]`: List of Job objects
+        # Raises
+            `RestAPIError`: If unable to get the jobs
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "jobs",
+        ]
+        query_params = {"expand": ["creator"]}
+        return job.Job.from_response_json(
+            _client._send_request("GET", path_params, query_params=query_params)
+        )
+
+    def exists(self, name: str):
+        """Check if a job exists.
+
+        # Arguments
+            name: Name of the job.
+        # Returns
+            `bool`: True if the job exists, otherwise False
+        # Raises
+            `RestAPIError`: If unable to check the existence of the job
+        """
+        try:
+            self.get_job(name)
+            return True
+        except RestAPIError:
+            return False
+
+    def get_configuration(self, type: str):
+        """Get configuration for the specific job type.
+
+        # Arguments
+            type: Type of the job. Currently, supported types include: SPARK, PYSPARK, PYTHON, DOCKER, FLINK.
+        # Returns
+            `dict`: Default job configuration
+        # Raises
+            `RestAPIError`: If unable to get the job configuration
+        """
+        _client = client.get_instance()
+        path_params = [
+            "project",
+            _client._project_id,
+            "jobs",
+            type.lower(),
+            "configuration",
+        ]
+
+        headers = {"content-type": "application/json"}
+        return _client._send_request("GET", path_params, headers=headers)
+
+    def _delete(self, job):
```

Comment:
Why is this one hidden vs the others?

Reply:
I am not sure, but it was this way in the original code.